harness/determined/common/experimental/trial.py

```diff
@@ -165,59 +195,95 @@ def select_checkpoint(
             resp = bindings.get_GetCheckpoint(self._session, checkpointUuid=uuid)
             return checkpoint.Checkpoint._from_bindings(resp.checkpoint, self._session)
 
-        def get_one(offset: int) -> bindings.v1GetTrialCheckpointsResponse:
+        order_by = None
+        if latest:
+            sort_by = CheckpointSortBy.BATCH_NUMBER
+            order_by = CheckpointOrderBy.DESC
+
+        if sort_by:
+            order_by = CheckpointOrderBy.ASC if smaller_is_better else CheckpointOrderBy.DESC
+
+        checkpoints = self.get_checkpoints(sort_by=sort_by, order_by=order_by)
+
+        if not checkpoints:
+            raise ValueError("No checkpoints found for criteria.")
+        return checkpoints[0]
+
+    def get_checkpoints(
```

Comment:
Small feature request: can we add a`skip_deleted: bool` or similar flag to `get_checkpoints` which filters out any deleted checkpoints from the returned list? I wouldn't usually care to see them in the returned list and it would save the users from filtering out manually.

Reply:
i considered this as well, but thought it might have been too specific to the particular checkpoint GC use case. though i can see how for all users, DELETED checkpoints might have limited value. i think a bool is definitely too specific, but maybe something more generic like `status=List[CheckpointState]` could be a potential filter parameter? 

my thought here is that if we support filtering, we should probably support more generic filtering. if we do that for other non-discrete parameters (reporting time, batch count, etc.) the API becomes a bit muddy. practically this is a one-liner for someone to implement:

`filter(lambda x: x.state != CheckpointState.DELETED, checkpoints)`

Reply:
yeah, that's fair. I also wouldn't want `CheckpointState.ERROR` cases, either. Guess I'd actually want a `completed_only` flag.

Easy to add something along these lines later, if requested.